src/wmtk/simplex/cofaces_single_dimension.cpp

```diff
@@ -33,6 +33,28 @@ std::vector<Tuple> cofaces_single_dimension_tuples(
     return {};
 }
 
+std::vector<Simplex> cofaces_single_dimension_simplices(
+    const Mesh& mesh,
+    const Simplex& simplex,
+    PrimitiveType cofaces_type)
+{
+    switch (mesh.top_simplex_type()) {
```

Comment:
there is a helper function to convert avector of tuples and a primitive type to a vector of simplices. use that API rather than all of this copy pasting. the tuples implementation should be slightly more efficient and hsould be the "true" implementation